pkg/security/security_profile/activity_tree/activity_tree.go

```diff
@@ -917,3 +917,16 @@ func (at *ActivityTree) ExtractPaths() (map[string][]string, map[string][]string
 
 	return fimPathsperExecPath, execAndParent
 }
+
+// ExtractSyscalls return the syscalls present in an activity tree
+func (at *ActivityTree) ExtractSyscalls() []string {
+	var syscalls []string
+
+	at.visit(func(processNode *ProcessNode) {
+		for _, s := range processNode.Syscalls {
+			syscall := model.Syscall(s).ConvertSyscallName()
+			syscalls = append(syscalls, syscall)
+		}
+	})
+	return syscalls
+}
```

Comment:
I wonder if maybe you could use at.SyscallsMask instead (and/or use the ComputeSyscallsList() method)?